protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java

```diff
@@ -0,0 +1,85 @@
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.api;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.james.imap.processor.IdleProcessor;
+
+import com.google.common.collect.ImmutableSet;
+
+public class ImapConfiguration {
+    public static Builder builder() {
+        return new Builder();
+    }
+    public static class Builder {
+        private long idleTimeInterval;
+        private TimeUnit idleTimeIntervalUnit;
+        private Set<String> disabledCaps;
+
+        private Builder() {
+            this.idleTimeInterval = IdleProcessor.DEFAULT_HEARTBEAT_INTERVAL_IN_SECONDS;
+            this.idleTimeIntervalUnit = IdleProcessor.DEFAULT_HEARTBEAT_INTERVAL_UNIT;
+            this.disabledCaps = ImmutableSet.of();
+        }
+
+        public Builder idleTimeInterval(long idleTimeInterval) {
+            this.idleTimeInterval = idleTimeInterval;
+            return this;
+        }
+
+        public Builder idleTimeIntervalUnit(TimeUnit idleTimeIntervalUnit) {
+            this.idleTimeIntervalUnit = idleTimeIntervalUnit;
+            return this;
+        }
+
+        public Builder disabledCaps(Set<String> disabledCaps) {
+            this.disabledCaps = disabledCaps;
+            return this;
+        }
+
+        public ImapConfiguration build() {
+            return new ImapConfiguration(idleTimeInterval, idleTimeIntervalUnit, disabledCaps);
+        }
+    }
+
+    private final long idleTimeInterval;
+    private final TimeUnit idleTimeIntervalUnit;
+    private final Set<String> disabledCaps;
+
+    private ImapConfiguration(long idleTimeInterval, TimeUnit idleTimeIntervalUnit, Set<String> disabledCaps) {
+        this.idleTimeInterval = idleTimeInterval;
+        this.idleTimeIntervalUnit = idleTimeIntervalUnit;
+        this.disabledCaps = disabledCaps;
+    }
+
+    public long getIdleTimeInterval() {
+        return idleTimeInterval;
+    }
+
+    public TimeUnit getIdleTimeIntervalUnit() {
+        return idleTimeIntervalUnit;
+    }
+
+    public Set<String> getDisabledCaps() {
+        return disabledCaps;
+    }
```

Comment:
> The copy should be done in the Builder.build() method.

I totally change the type of disablesCap to ImmutableSet. I think it makes sense when it loaded one any only one time from config file then it will not modify anymore. So it does not need copy anymore. Is that OK?